lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -382,11 +400,32 @@ def _BuildInserts(self, values):
 
     return [aff4_q, subjects_q, attributes_q]
 
-  def _ExecuteTransaction(self, transaction):
+  def _ExecuteQuery(self, query, args=None):
     """Get connection from pool and execute query."""
+    retries = 10
+    for attempt in range(1, retries + 1):
+      connection = self.pool.GetConnection()
+      try:
+        connection.cursor.execute(query, args)
```

Comment:
This whole thing could do with lots of stats.STAT counters and timers added to enable proper monitoring and give you visibility of query execution times, error rates, retries, aborts etc.